From f627dd7b4de75c8f107983b7f43f0a9c64faa388 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 28 Apr 2022 19:47:17 +0300
Subject: [PATCH] [SW-82716] gaudi3: initialize i2c register structure

This patch adds initialization of i2c register which will be
performed as part of recovery init.

Change-Id: I2e39e9dbb3edfa8589116016f9ac62cb2a9c59fa
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 52dcf62..f837228 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -167,11 +167,38 @@
 {
 	global_conf.upd_cmd_map = mmUPD_CMD;
 	global_conf.upd_sts_map = mmUPD_STS;
+	global_conf.i2c_slv_reg = mmD0_PSOC_GLOBAL_CONF_I2C_SLV;
+	global_conf.i2c_slv_intr_mask_reg = mmD0_PARC_INT_AGGR_I2C_SLV_COMB_MASK;
+}
+
+static void device_init_i2c_s(void)
+{
+	i2c_s.enable_reg = mmD0_PSOC_I2C_S_IC_ENABLE;
+	i2c_s.sar_reg = mmD0_PSOC_I2C_S_IC_SAR;
+	i2c_s.con_reg = mmD0_PSOC_I2C_S_IC_CON;
+	i2c_s.intr_mask_reg = mmD0_PSOC_I2C_S_IC_INTR_MASK;
+	i2c_s.intr_stat_reg = mmD0_PSOC_I2C_S_IC_INTR_STAT;
+	i2c_s.rx_tl_reg = mmD0_PSOC_I2C_S_IC_RX_TL;
+	i2c_s.tx_tl_reg = mmD0_PSOC_I2C_S_IC_TX_TL;
+	i2c_s.data_cmd_reg = mmD0_PSOC_I2C_S_IC_DATA_CMD;
+	i2c_s.clr_tx_abrt_reg = mmD0_PSOC_I2C_S_IC_CLR_TX_ABRT;
+	i2c_s.clr_rx_under_reg = mmD0_PSOC_I2C_S_IC_CLR_RX_UNDER;
+	i2c_s.clr_rx_over_reg = mmD0_PSOC_I2C_S_IC_CLR_RX_OVER;
+	i2c_s.clr_tx_over_reg = mmD0_PSOC_I2C_S_IC_CLR_TX_OVER;
+	i2c_s.clr_rx_done_reg = mmD0_PSOC_I2C_S_IC_CLR_RX_DONE;
+	i2c_s.clr_activity_reg = mmD0_PSOC_I2C_S_IC_CLR_ACTIVITY;
+	i2c_s.clr_stop_det_reg = mmD0_PSOC_I2C_S_IC_CLR_STOP_DET;
+	i2c_s.clr_start_det_reg = mmD0_PSOC_I2C_S_IC_CLR_START_DET;
+	i2c_s.clr_gen_call_reg = mmD0_PSOC_I2C_S_IC_CLR_GEN_CALL;
+	i2c_s.clr_rd_req_reg = mmD0_PSOC_I2C_S_IC_CLR_RD_REQ;
+	i2c_s.clr_intr_reg = mmD0_PSOC_I2C_S_IC_CLR_INTR;
+	i2c_s.rxflr_reg = mmD0_PSOC_I2C_S_IC_RXFLR;
 }
 
 static void device_init(void)
 {
 	device_init_global_conf();
+	device_init_i2c_s();
 }
 
 static void rom_recovery(void)
